### ISSUE TITLE
PartialUpdateUserAsync behaves differently in Okta.Sdk v6 

### ISSUE THREAD

**Issue by LanceXact** [okta/okta-sdk-dotnet#614]
### Describe the bug?

In Okta.Sdk v6, providing a partially filled UserProfile to PartialUpdateUserAsync behaves differently than Okta.Sdk v5.

### What is expected to happen?

Properties that are not set on the UserProfile model should not be sent in the partial update request.

### What is the actual behavior?

Properties not set on the UserProfile model are sent with a value of "null" causing fields to undesirably get set to null in Okta, or causing a validation error.

### Reproduction Steps?

**In Okta.Sdk v6, run the following code:**

```c#
await userApi.PartialUpdateUserAsync(userToUpdate.Id, new UpdateUserRequest
{
    Profile = new UserProfile
    {
        PrimaryPhone = "321-123-1000",
    },
});
```

You will receive a 400 bad request response. This is because firstName and lastName are being sent as null in the request, even though they were not specified. Here is an example of the request:
```
POST https://<your_okta_domain>/api/v1/users/<your_user_id> HTTP/1.1
Host: <your_okta_domain>
Accept: application/json
Authorization: SSWS <your_api_token>
User-Agent: Okta.Sdk/6.0.2 runtime/.NET 7.0.2 os/Microsoft Windows 10.0.19044 /okta-sdk-dotnet/csharp/oasv3
Connection: Keep-Alive
Accept-Encoding: gzip, deflate
Content-Type: application/json
Content-Length: 215

{"profile":{"city":null,"countryCode":null,"firstName":null,"lastName":null,"mobilePhone":null,"postalAddress":null,"primaryPhone":"321-123-1000","secondEmail":null,"state":null,"streetAddress":null,"zipCode":null}}
```

**In Okta.Sdk v5, run the following code:**

```c#
var oktaUser = new User();
oktaUser.SetProperty("id", userToUpdate.Id);
oktaUser.Profile = new UserProfile
{
    PrimaryPhone = "321-123-1000",
};
await oktaClient.Users.PartialUpdateUserAsync(oktaUser, oktaUser.Id);
```
This will succeed. This is because only the "id" and "primaryPhone" fields are sent in the request. Here is an example of the request:
```
POST https://<your_okta_domain>/api/v1/users/<your_okta_user> HTTP/1.1
Host: <your_okta_domain>
User-Agent: okta-sdk-dotnet/5.6.2 runtime/.NET 7.0.2 os/Microsoft Windows 10.0.19044
Authorization: SSWS <your_api_token>
Cookie: JSESSIONID=315B44B358062169DB48C0F85D6F2DE6
Content-Type: application/json; charset=utf-8
Content-Length: 71

{"id":"00u863qakrc2nQwGI5d7","profile":{"primaryPhone":"321-123-1000"}}
```

### Additional Information?

I created a repository showcasing the difference:
https://github.com/nobowned/okta-sdk-partial-update-bug/blob/main/OktaPartialUpdateBug/OktaPartialUpdateBug/Program.cs

The "OKTA_SDK_6" compilation symbol can be specified or left out to switch between the two Okta.Sdk versions.

### .NET Version

7.0.102

### SDK Version

6.0.2

### OS version

BuildNumber  Caption                          OSArchitecture  Version
19044        Microsoft Windows 10 Enterprise  64-bit          10.0.19044

**Comment by laura-rodriguez** [comment]
Hi @LanceXact,

Thanks for reporting this issue. I'll file an internal ticket to be reviewed and prioritized by the team.

Internal Ref: OKTA-573429

**Comment by laura-rodriguez** [comment]
Hi @LanceXact,

I've just added an integration test with your sample code to reproduce the bad request error, but the test is running green. You can take a look [here](https://github.com/okta/okta-sdk-dotnet/pull/619). Can you please let me know if I'm missing something?

**Comment by laura-rodriguez** [comment]
Fixed in 6.0.4
